batch/batch/batch.py

```diff
@@ -93,7 +97,7 @@ def copy_command(src, dst):
                 return f'{mkdirs} gsutil -m cp -R {shq(src)} {shq(dst)}'
 
             copies = ' && '.join([copy_command(src, dst) for (src, dst) in files])
-            sh_expression = f'{authenticate} && {copies}'
+            sh_expression = f'{check_success} || {clean} && {authenticate} && {copies} && {touch_success}'
```

Comment:
Hrm. I find this a bit confusing and I think `set -ex` is in the wrong spot. Does the following look OK to you? I think I find it more clear.

```bash
authenticate = 'gcloud ...'
success_file = '/io/__BATCH_{}_SUCCESS__'.format(task_name.upper())
clean_if_necessary = 'rm -rf /io/*' if task_name == 'input' else 'true'
sh_expression = f"""set -ex
if [ -e {success_file} ]; then
    exit 0
fi
{clean_if_necessary} && {authenticate} && {copies} && {touch_success}
"""
```

Reply:
Also, with `set -ex` we don't need the ampersands, if any of those commands fail it all fails, so we can just end it with:
```bash
{clean_if_necessary}
{authenticate}
{copies}
{touch_success}
```